be/src/storage/rowset/rowset_factory.cpp

```diff
@@ -37,9 +37,9 @@ Status RowsetFactory::create_rowset(const TabletSchema* schema, const std::strin
     if (rowset_meta->rowset_type() == BETA_ROWSET) {
         *rowset =
                 BetaRowset::create(ExecEnv::GetInstance()->tablet_meta_mem_tracker(), schema, rowset_path, rowset_meta);
-        return (*rowset)->init() == OLAP_SUCCESS ? Status::OK() : Status::InternalError("fail to init rowset");
+        RETURN_IF_ERROR((*rowset)->init());
     }
-    return Status::NotSupported("unsupported rowset type");
+    return Status::OK();
```

Comment:
OK() ?

Reply:
This place is right because the ERROR is returned already in line 40.